Haohui - Gives the user permission to manage individual user permissions.

Lin1404 opened the pull request:
# Description
Please include the exact bug/functionality description and a summary of the changes/ related issues. Please also include any other relevant motivation and context:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78375725/20b58dab-a869-44ba-91de-147587525a60)

## Related PRS (if any):
This frontend PR is related to the backend PR # 532.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as owner user call user A
4. go to dashboard→ Other Links→ Permissions Management→ Manage User Permissions
5. input another user name, call user B
6. find the "Manage User Permissions", click the add button
7. scroll down and click submit button
8. log out and login as user B
9. go to dashboard→ Other Links→ Permissions Management→ Manage User Permissions
10. check the user B able to manage user permissions.

## Screenshots or videos of changes:
https://drive.google.com/file/d/1gEo7ixKdH1-o3ZnFts0N_L_Hsm73m34M/view?usp=sharing

## Note:
Include the information the reviewers need to know.

GaryB93 commented on src/components/PermissionsManagement/UserPermissionsPopUp.jsx:
```diff
@@ -43,6 +43,7 @@ const UserPermissionsPopUp = ({ allUserProfiles, toggle, getAllUsers, roles }) =
       const newPermissionsObject = {
         frontPermissions: actualPermissionsFront,
       };
+      console.log(newPermissionsObject.frontPermissions)
```
@Lin1404 Can we remove the console.log?

Lin1404 replied:
Good catch! Thank you Gary!

GaryB93 reviewed (dismissed):
Great work on implementing the new permission functionality. I was able to add the Manage User Permissions on a Volunteer role and that user account was only allowed to manage user permissions on the User Roles page and was restricted from the rest of the functionality on that page.
![Screenshot 2023-09-11 at 3 20 11 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108501006/1af93031-52a9-46c0-86c6-02b7185b445b)

robertoooc reviewed (approved):
Hi @Lin1404, I tested your pr along with the backend implementation as well and it performed as intended. Didn't catch any issues while testing it as well. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/117053202/718b3ef0-4670-4e46-ba97-f393aeb4deff

AriaYu927 commented:
Hi @Lin1404 , I've tested your PR and it works as expected. Thank you!

Before added permission to YIhantest Core:
<img width="1173" alt="截屏2023-09-13 13 25 33" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/cdda5822-1056-4cda-bd99-fb5c81d98c5e">

After added permission:
<img width="1173" alt="截屏2023-09-13 13 26 43" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/8472e2e5-265b-4932-a389-b1e6841fdfc6">

GaryB93 commented on src/routes.js:
```diff
@@ -123,7 +123,7 @@ export default (
             exact
             component={PermissionsManagement}
-            routePermissions={RoutePermissions.permissionsManagement}
+            routePermissions={[RoutePermissions.permissionsManagement,RoutePermissions.userPermissionsrManagement]}
```
Hey @Lin1404, is there supposed to be a 'r' between Permissions and Management on `RoutePermissions.userPermissionsrManagement` ?

GaryB93 reviewed (dismissed):
Hey @Lin1404, tested this updated PR with backend PR#532 to verify the permission of Manage User Permissions can be given to an otherwise unauthorized user and would function as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108501006/abde0258-7ca5-4f5a-a3c3-398f00d378d9

wantingxu7 commented:
Hi @Lin1404 , your work works as expected in my local machine, good job!
![User B_before given permission](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/5f924aee-cca2-4c27-9475-c941075
![User B_after given permission_p1](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/74728a2e-5dc2-4e72-911e-dce4feac13f4)
f988e)
![User B_after given permission_p2](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/1efbe2b6-9491-47cb-b37e-b12d02699695)

Shiwani99 reviewed (changes requested):
Hi @Lin1404,
I tested your PR and I wanted to add a problem that I faced. When I set user management permissions from owner to manager I'm able to see user management in manager, but when I set permission for volunteer from manager, I'm not able to see the permissions get saved after I hit submit in manager profile because of which I'm not able to see other links in volunteer page.  This is also happening for one other volunteer profile I created where in I removed my user management permission from manager to another volunteer but it is not getting removed despite giving submit. I tried setting permissions for test account too from manager, but I'm encountering the same issue. There is no issue when I try to add or remove user management permission for any user form owner account.
I'm attaching a screen recording of what is happening. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/c9bf0365-f67b-4828-9e1e-6edb2d206ac0

DavidC0126 reviewed (approved):
Hi @Lin1404 I tested your PR and it works as described.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/312c67b3-6908-4c95-9aa7-1f7f69e6f608

Lin1404 commented:
> Hi @Lin1404 I tested your PR and it works as described.
> 
>  permission_user.mov

Hi @DavidC0126 I retested it and I think the problem is fixed now.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78375725/61f0182e-b9a5-48ca-9980-fdb51290efd5

Jiyuanxia reviewed (approved):
Hi @Lin1404 , I have reviewed your pull request and tested it. It works as you intended. Well done!
![Screenshot 2023-09-21 222357](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92963742/70a25574-80dc-4bcb-8142-00341c2c2ac7)
![Screenshot 2023-09-21 222217](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92963742/969fb754-77a3-4ab2-be9c-4f7e4f6d3b3b)

harshidad reviewed (approved):
I tested the PR with backend PR#532 to verify the permission of users and it's working as expected. Great Work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/23d0063b-7ce8-46b1-aa97-81b632066bba

belagaam reviewed (approved):
I tested the PR as per the instructions and everything worked as expected.




















https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47104198/6681a12e-2c23-4ccf-a1c1-4a8e34c4071c

EdwinLauM reviewed (approved):
Hey @Lin1404! Works as expected, good job! 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/900359c5-5317-45d0-8a70-c3fc3db73e39

Lin1404 commented:
> Hi @Lin1404, I tested your PR and I wanted to add a problem that I faced. When I set user management permissions from owner to manager I'm able to see user management in manager, but when I set permission for volunteer from manager, I'm not able to see the permissions get saved after I hit submit in manager profile because of which I'm not able to see other links in volunteer page. This is also happening for one other volunteer profile I created where in I removed my user management permission from manager to another volunteer but it is not getting removed despite giving submit. I tried setting permissions for test account too from manager, but I'm encountering the same issue. There is no issue when I try to add or remove user management permission for any user from owner account. I'm attaching a screen recording of what is happening.
> 
>  Screen.Recording.2023-09-14.at.6.01.38.PM.mov

Hi @Shiwani99 , the reason the bug happen because the permission type different. I have fixed and retest it, I believe it works well now. Please check out the retest video below.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78375725/61f0182e-b9a5-48ca-9980-fdb51290efd5

Shiwani99 reviewed (approved):
Hi,
Thank you for the changes. Great work. I retested it and the bug was resolved now. It works as expected. I'm attaching screen recording of my test.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/4f6f1ebc-e94e-4813-ba05-ee272c1cff71

one-community commented:
Thank you all, moving to final review

emunkh404 commented:
Nice work
